lda/tweet_fetcher.py

```diff
@@ -0,0 +1,89 @@
+import os
+import random
+import time
+
+# http://docs.tweepy.org/en/latest/index.html
+import tweepy
+
+from lda.twitter_secret_fetcher import get_api_key, get_api_secret_key, get_access_token, get_access_token_secret
+
+# Percentage of COVID tweets to sample, there are 239,861,658 in total
+SAMPLE_PERCENTAGE = 0.0001
+# Directory containing files that have COVID tweet IDs
+# I left this out of the git repo because it takes a LONG time to push and pull. So if you want to run this, just
+# download it your self and place them in this directory
+COVID_TWEET_ID_DIR = "../data/covid_tweet_ids"
+# Directory containing files that have COVID tweet text
+COVID_TWEET_TEXT_DIR = "../data/covid_tweets"
+
+
+###############################################
+# Sample Tweet Ids From File
+###############################################
+
+# Tweet IDs come from here: https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/LW0BTB
+def sample_tweet_ids():
+    sampled_ids = []
+    for file_name in os.listdir(COVID_TWEET_ID_DIR):
+        sampled_ids.extend(sample_tweet_ids_from_file(file_name))
+    return sampled_ids
+
+
+def sample_tweet_ids_from_file(file_name):
+    # We have to read through every line which could be time consuming. If this becomes a bottleneck look into algo #3
+    # from here: http://metadatascience.com/2014/02/27/random-sampling-from-very-large-files/ or some other faster algo
+    sampled_ids = []
+    with open(f"../data/covid_tweet_ids/{file_name}") as f:
+        for tweet_id in f.readlines():
+            if should_sample():
+                sampled_ids.append(int(tweet_id.strip()))
+    print(f"Sampled {file_name}")
+    return sampled_ids
+
+
+def should_sample():
+    return random.random() < SAMPLE_PERCENTAGE
+
+
+###############################################
+# Get Tweets From Twitter API
+###############################################
+
+def connect_to_twitter():
+    auth = tweepy.OAuthHandler(get_api_key(), get_api_secret_key())
+    auth.set_access_token(get_access_token(), get_access_token_secret())
+    return tweepy.API(auth)
+
+
+def get_tweets(tweet_ids, api):
+    ts = time.time()
```

Comment:
Does getting the current time here mean that we are grabbing the SAMPLE_PERCENTAGE of total tweets starting from now and moving backwards until the limit is reached?

Reply:
No we're just using the current timestamp as a unique identifier when creating the file so we don't accidently delete another file. There's no logic currently to get tweets from a certain time range, we just take some percentage of all tweet IDs. I can probably clarify this in a comment.